TESTS/mbed_drivers/lp_ticker/main.cpp

```diff
@@ -0,0 +1,259 @@
+   int32_t ret;
+
+   timer.start();
+   ticker.attach_us(callback(wait_callback_sem), 100000);
+   ret = ticker_callback_sem.wait();
+   timer.stop();
+   int time_diff = timer.read_us();
+   TEST_ASSERT_UINT32_WITHIN((uint32_t)TOLERANCE, 100000, time_diff);
+   ...
+
+   For 100000[us] callback for NUCLEO_F070RB we get time_diff: 100819[us]
+   So the extra 819[us] is from lowpower timer inaccuracy and time measure functions calls
+
+   @note for LowPowerTicker the tolerance value is 6 times more than for Ticker due to the
+   less ticker precision
+
+   To take in to account this extra time we introduces TOLERANCE value based on CPU clock (speed)
+
+   e.g.
+   For K64F          TOLERANCE = (60000 / 120000000) * 1000000 =  500[us]
+   For NUCLEO_F070RB TOLERANCE = (60000 /  48000000) * 1000000 = 1250[us]
+   For NRF51_DK      TOLERANCE = (60000 /  16000000) * 1000000 = 3750[us]
+ */
+#define TOLERANCE ((TOLERANCE_FACTOR / (float)SystemCoreClock) * US_FACTOR)
+
+#define TICKER_COUNT 16
+
+Semaphore ticker_callback_sem(0, 1);
+bool ticker_callback_flag;
+volatile uint32_t multi_counter;
+
+
+void wait_callback_sem(void)
+{
+    ticker_callback_flag = true;
+    ticker_callback_sem.release();
+}
+
+void wait_callback_multi(void)
+{
+    multi_counter++;
+}
+
+
+/** Test many tickers run simultaneously
+
+    Given an 16 tickers
+    When schedule them
+    Then tickers properly execute callbacks
+ */
+void test_multi_ticker(void)
+{
+    TestTicker ticker[TICKER_COUNT];
+
+    multi_counter = 0;
+    for(int i = 0; i < TICKER_COUNT; i++) {
+        ticker[i].attach_us(callback(wait_callback_multi), 100 * 1000);
+    }
+
+    Thread::wait(110);
+    for(int i = 0; i < TICKER_COUNT; i++) {
+            ticker[i].detach();
+    }
+
+    //TEST_ASSERT_EQUAL(TICKER_COUNT, multi_counter);
+
+    multi_counter = 0;
+    for(int i = 0; i < TICKER_COUNT; i++) {
+        ticker[i].attach_us(callback(wait_callback_multi), (100 + i) * 1000);
+    }
+
+    Thread::wait(130);
+    for(int i = 0; i < TICKER_COUNT; i++) {
+        ticker[i].detach();
+    }
+    TEST_ASSERT_EQUAL(TICKER_COUNT, multi_counter);
+}
+
+/** Test multi callback time
+
+    Given an Ticker
+    When schedule callback
+    Then ticker properly execute callback multiple time
+ */
+void test_multi_call_time(void)
+{
+    TestTicker ticker;
+    int32_t ret;
+    Timer timer;
+    int time_diff;
+
+    for(int i = 0; i < 10; i++) {
+        ticker_callback_flag = false;
+        timer.reset();
+
+        timer.start();
+        ticker.attach_us(callback(wait_callback_sem), 100000);
+        ret = ticker_callback_sem.wait();
+        timer.stop();
+        time_diff = timer.read_us();
+
+        TEST_ASSERT_EQUAL(true, ticker_callback_flag);
+        TEST_ASSERT_TRUE(ret > 0);
+        TEST_ASSERT_UINT32_WITHIN(TOLERANCE, 100000, time_diff);
+    }
+}
+
+/** Test if detach cancel scheduled callback event
+
+    Given an Ticker with scheduled callback
+    When schedule is cancelled
```

Comment:
schedule is not really something that exists in Ticker terms, I know what you mean, but we could use something like: 
Given a Ticker with callback attached
When the callback is detached
Then the callback is not being called

Also your test does more than described here.